xdsl/parser/attribute_parser.py

```diff
@@ -276,6 +276,16 @@ def _parse_dialect_type_or_attribute_body(
         else:
             raise TypeError("Attributes are either ParametrizedAttribute or Data.")
 
+    @overload
+    def _parse_extended_type_or_attribute(
+        self, attr_or_dialect_name: str, is_type: Literal[False]
+    ) -> Attribute: ...
+
+    @overload
+    def _parse_extended_type_or_attribute(
+        self, attr_or_dialect_name: str, is_type: Literal[True] = True
```

Comment:
Do you need the default here?

Reply:
I'm not sure whether it was needed originally, but since this is a private method and this is actually an important argument to get right, I removed the default value, which seems to work fine.